packages/main/src/plugin/context/context.ts

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// based on https://github.com/microsoft/vscode/blob/3eed9319874b7ca037128962593b6a8630869253/src/vs/platform/contextkey/browser/contextKeyService.ts
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { ApiSenderType } from '../api.js';
+import type { IContext } from '../api/context-info.js';
+
+export class Context implements IContext {
+  private _value: Record<string, any>;
+
+  constructor(
+    private _id: number,
+    private _parent: Context | null,
```

Comment:
from my understanding parent and extension are optional so they should be at the end of the parameters

also probably same than before, probably it's optional so undefined and not null